include/nbl/builtin/hlsl/bxdf/common.hlsl

```diff
@@ -15,30 +17,74 @@ namespace hlsl
 namespace bxdf
 {
 
+// returns unnormalized vector
+// TODO: template these?
+float computeUnnormalizedMicrofacetNormal(bool _refract, float3 V, float3 L, float orientedEta)
+{
+    const float etaFactor = (_refract ? orientedEta : 1.0);
+    const float3 tmpH = V + L * etaFactor;
+    return _refract ? (-tmpH) : tmpH;
```

Comment:
btw you could make a function somewhere (to be moved into the @Przemog1 ieee754 namespace) that conditionally flips the sign of a float